synapse/events/utils.py

```diff
@@ -130,6 +130,16 @@ def add_fields(*fields: str) -> None:
         add_fields("membership")
         if room_version.msc3375_redaction_rules:
             add_fields(EventContentFields.AUTHORISING_USER)
+        if room_version.msc3821_redaction_rules:
+            # Preserve the signed field under third_party_invite.
+            third_party_invite = event_dict["content"].get("third_party_invite")
+            if isinstance(third_party_invite, collections.abc.Mapping):
+                new_content["third_party_invite"] = {}
+                if "signed" in third_party_invite:
+                    new_content["third_party_invite"]["signed"] = third_party_invite[
+                        "signed"
+                    ]
```

Comment:
Looks reasonable. We may want to consider whether to do the same empty object and only specific `signed` sub-fields like [MSC3389](https://github.com/matrix-org/matrix-spec-proposals/pull/3389) and https://github.com/matrix-org/synapse/pull/15565

But it does what the MSC says atm :+1:

Reply:
Yeah the MSC says *not* to do that. (I don't like that the two MSCs say different things... I'll leave a comment on one of them.)

Reply:
I left a comment on MSC3389, see https://github.com/matrix-org/matrix-spec-proposals/pull/3389/files#r1193739511